test/test_t8code_3d.jl

```diff
@@ -0,0 +1,249 @@
+module TestExamplesT8codeMesh3D
+
+using Test
+using Trixi
+
+include("test_trixi.jl")
+
+EXAMPLES_DIR = joinpath(examples_dir(), "t8code_3d_dgsem")
+
+# Start with a clean environment: remove Trixi.jl output directory if it exists
+outdir = "out"
+isdir(outdir) && rm(outdir, recursive = true)
+mkdir(outdir)
+
+@testset "T8codeMesh3D" begin
+    @trixi_testset "elixir_advection_basic.jl" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_basic.jl"),
+                            # Expected errors are exactly the same as with TreeMesh!
+                            l2=[0.00016263963870641478],
+                            linf=[0.0014537194925779984])
+    end
+
+    @trixi_testset "elixir_advection_unstructured_curved.jl" begin
+        @test_trixi_include(joinpath(EXAMPLES_DIR,
+                                     "elixir_advection_unstructured_curved.jl"),
+                            l2=[0.0004750004258546538],
+                            linf=[0.026527551737137167])
```

Comment:
Could you please add comments that the errors are expected to be the same as with other mesh types whenever appropriate? IIRC, you said that basically all new examples are adaptations of existing examples.

Reply:
Done.